x-pack/plugins/apm/public/components/shared/environment_select/index.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React, { useState, useCallback } from 'react';
+import { throttle } from 'lodash';
+import { EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { getEnvironmentLabel } from '../../../../common/environment_filter_values';
+import { useEnvironmentCustomOptions } from '../../../hooks/use_environment_custom_options';
+import { FETCH_STATUS } from '../../../hooks/use_fetcher';
+import { useEnvironmentsContext } from '../../../context/environments_context/use_environments_context';
+
+export function EnvironmentSelect({
+  onChange,
+}: {
+  onChange: (value?: string) => void;
```

Comment:
I think it's confusing to have this component emit an onChange event, but not have the environment value as a prop. Ideally it is either a controlled component, taking both an `environment` or `value` prop, and an `onChange` prop, or uncontrolled, having neither.